### PR TITLE
toolkit/pkgbld.sh: add a helper script to build packages locally

### PR DESCRIPTION
**Pull request by mfrw**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] If you are adding/removing a .spec file that has multiple-versions supported, please add [@microsoft/cbl-mariner-multi-package-reviewers](https://github.com/orgs/microsoft/teams/cbl-mariner-multi-package-reviewers) team as reviewer [(Eg. golang has 2 versions 1.18, 1.21+)](https://github.com/microsoft/azurelinux/tree/2.0/SPECS/golang)
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Add a helper script to simply building locally. It has sane defaults.

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- NA

###### Links to CVEs  
- NA

###### Test Methodology

- Local Build.

### CODE REVIEW

File: toolkit/build.sh [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	PACKAGE_BUILD_LIST="${TOBUILD}" \
+	PACKAGE_REBUILD_LIST="${TOBUILD}" \
+	SRPM_PACK_LIST="${TOBUILD}" \
+	RUN_CHECK=$RC \
+	SRPM_FILE_SIGNATURE_HANDLING=update \
+	DAILY_BUILD_ID=$BID \
+	REFRESH_WORKER_CHROOT=$WCREFRESH \
+	USE_PACKAGE_BUILD_CACHE=$USEPKGBLDCACHE \
+	REPO_LIST=$REPOLIST \
+	SPECS_DIR=$SDIR
```
Comment (dmcilvaney) [inline toolkit/build.sh:131]: >= 2 massively speeds up caching and spec parsing for large builds (less important here), but large values of -j really speed up the toolchain downloader since each rpm can go in parallel.
```suggestion
	SPECS_DIR=$SDIR \
	-j 30
```

File: toolkit/build.sh [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+fi
+
+
+
+
+sudo make build-packages \
+	REBUILD_TOOLS=y \
+	SOURCE_URL="https://azurelinuxsrcstorage.blob.core.windows.net/sources/core" \
+	PACKAGE_BUILD_LIST="${TOBUILD}" \
+	PACKAGE_REBUILD_LIST="${TOBUILD}" \
```
Comment (dmcilvaney) [inline toolkit/build.sh:123]: Only need one or the other if the lists are the same, they get unioned.
```suggestion
	PACKAGE_REBUILD_LIST="${TOBUILD}" \
```

File: toolkit/build.sh [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+	sudo make clean-expand-specs clean-input-srpms
+fi
+
+if [ -n "$dryrunVar" ]
+then
+	echo "sudo make build-packages \ "
+	echo "     REBUILD_TOOLS=y \ "
+	echo "     SOURCE_URL="https://azurelinuxsrcstorage.blob.core.windows.net/sources/core" \ ";
+	echo "     PACKAGE_BUILD_LIST="${TOBUILD}" \ ";
+	echo "     PACKAGE_REBUILD_LIST="${TOBUILD}" \ ";
```
Comment (dmcilvaney) [inline toolkit/build.sh:105]:
```suggestion
	echo "     PACKAGE_REBUILD_LIST="${TOBUILD}" \ ";
```

File: toolkit/build.sh [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+	echo "     SRPM_PACK_LIST="${TOBUILD}" \ ";
+	echo "     RUN_CHECK=$RC \ ";
+	echo "     SRPM_FILE_SIGNATURE_HANDLING=update \ ";
+	echo "     DAILY_BUILD_ID=$BID \ ";
+	echo "     REFRESH_WORKER_CHROOT=$WCREFRESH \ ";
+	echo "     USE_PACKAGE_BUILD_CACHE=$USEPKGBLDCACHE \ ";
+	echo "     SPECS_DIR=$SDIR";
```
Comment (dmcilvaney) [inline toolkit/build.sh:112]:
```suggestion
	echo "     SPECS_DIR=$SDIR \ ";
	echo "     -j 30";
```

File: toolkit/build.sh [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+USEPKGBLDCACHE=y
+
+helpFuncion () {
+	echo ""
+	echo "-h  : Print detailed usage"
+        echo "-s  : Spec Dir [Default SPECS] [Possible values: ../SPECS-EXTENDED, ../SPECS]"
+        echo "-b  : DAILY_BUILD_ID of the format of 3-0-YYYYMMDD [Default 'lkg']"
+        echo "-c  : Run with check distabled/enabled [Default RUN_CHECK=n]"
+        echo "-p  : Package to build/rebuild [Default all]"
+        echo "-r  : Refresh worker chroot [Default false]"
```
Comment (dmcilvaney) [inline toolkit/build.sh:17]: I would invert `-r` to off by default. Most people won't be messing with the go tools (which is the main reason the chroot regenerates), and can lead to some weird inconsistencies if the toolchain packages get out of sync with the chroot.

File: toolkit/build.sh [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+then
+	echo "Cleaning up expanded specs and input srpms"
+	sudo make clean-expand-specs clean-input-srpms
+fi
+
+if [ -n "$dryrunVar" ]
+then
+	echo "sudo make build-packages \ "
+	echo "     REBUILD_TOOLS=y \ "
+	echo "     SOURCE_URL="https://azurelinuxsrcstorage.blob.core.windows.net/sources/core" \ ";
```
Comment (dmcilvaney) [inline toolkit/build.sh:103]:
```suggestion
```

File: toolkit/build.sh [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+	echo "     SPECS_DIR=$SDIR";
+	exit 0
+fi
+
+
+
+
+sudo make build-packages \
+	REBUILD_TOOLS=y \
+	SOURCE_URL="https://azurelinuxsrcstorage.blob.core.windows.net/sources/core" \
```
Comment (dmcilvaney) [inline toolkit/build.sh:121]:
```suggestion
```

File: toolkit/build.sh [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+
+helpFuncion () {
+	echo ""
+	echo "-h  : Print detailed usage"
+        echo "-s  : Spec Dir [Default SPECS] [Possible values: ../SPECS-EXTENDED, ../SPECS]"
+        echo "-b  : DAILY_BUILD_ID of the format of 3-0-YYYYMMDD [Default 'lkg']"
+        echo "-c  : Run with check distabled/enabled [Default RUN_CHECK=n]"
+        echo "-p  : Package to build/rebuild [Default all]"
+        echo "-r  : Refresh worker chroot [Default false]"
+        echo "-f  : Force a Rebuild"
```
Comment (dmcilvaney) [inline toolkit/build.sh:18]: `-f` seems redundant here since we are using `PACKAGE_REBUILD_LIST=...` which will always force rebuilds.

File: toolkit/build.sh [hunk 9]
```diff
@@ -0,0 +1,131 @@
+#!/bin/bash
+
```
Comment (dmcilvaney) [inline toolkit/build.sh:2]: Maybe a disclaimer that this is intended for local dev use, no warranty, subject to change etc. etc.

File: toolkit/build.sh [hunk 10]
```diff
@@ -0,0 +1,131 @@
+#!/bin/bash
```
Comment (neha170) [inline toolkit/build.sh:1]: Please add the MS copyright header

File: toolkit/build.sh [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+WCREFRESH=n
+SDIR=../SPECS
+USEPKGBLDCACHE=y
+
+helpFuncion () {
+	echo ""
+	echo "-h  : Print detailed usage"
+        echo "-s  : Spec Dir [Default SPECS] [Possible values: ../SPECS-EXTENDED, ../SPECS]"
+        echo "-b  : DAILY_BUILD_ID of the format of 3-0-YYYYMMDD [Default 'lkg']"
+        echo "-c  : Run with check distabled/enabled [Default RUN_CHECK=n]"
```
Comment (neha170) [inline toolkit/build.sh:15]: nit typo
`disabled`

File: toolkit/build.sh [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+fi
+
+if [ -n "$checkVar" ]
+then
+	RC=y
+fi
+
+if [ -z "$packageVar" ]
+then
+	echo "Package var needs to be set:";
```
Comment (neha170) [inline toolkit/build.sh:72]: I would reword this to 
`Package to build needs to be set`
to avoid looking up what Package _var_ refers to

File: toolkit/build.sh (hunk 11 above)
Comment (neha170) [inline toolkit/build.sh:15]: nit: to be consistent with the wording
`Default disabled` instead of `Default RUN_CHECK=n`

File: toolkit/build.sh (hunk 10 above)
Comment (neha170) [inline toolkit/build.sh:1]: I would consider renaming the script to `build_package.sh` to avoid the ambiguity on _what_ it is building.
Additionally, if we decide to add more such scripts in the future to build image/ toolchain, etc. we would not need to rename this script.

File: toolkit/build.sh (hunk 4 above)
Comment (neha170) [reply toolkit/build.sh:112]: Could use `-j $(nproc)` ?

File: toolkit/build.sh (hunk 9 above)
Comment (mfrw) [reply toolkit/build.sh]: sure ..

File: toolkit/build.sh (hunk 10 above)
Comment (mfrw) [reply toolkit/build.sh]: Ahh sure, added

File: toolkit/build.sh (hunk 12 above)
Comment (mfrw) [reply toolkit/build.sh]: Good catch.

File: toolkit/build.sh (hunk 10 above)
Comment (mfrw) [reply toolkit/build.sh]: I feel i might rename it to `pkgbld` so that it is not too much typing.

File: toolkit/build.sh (hunk 11 above)
Comment (mfrw) [reply toolkit/build.sh]: Ohh my spellings :(

File: toolkit/build.sh (hunk 7 above)
Comment (mfrw) [reply toolkit/build.sh]: Yes, this is the default.

File: toolkit/build.sh (hunk 5 above)
Comment (mfrw) [reply toolkit/build.sh]: Sure

**Review by neha170 — APPROVED** [review]

**Review by dmcilvaney — APPROVED** [review]
